tests/gnrc_tcp/tests/shared_func.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2018   Simon Brummer <simon.brummer@posteo.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import re
+import pexpect
+import random
+
+def drainTextBuffer(child):
+    output = ''
+    try:
+        while True:
+            output += child.read_nonblocking(size = 1, timeout = 0.5)
+
+    except pexpect.TIMEOUT:
+        pass
+
+    return output
+
+def getHostTapDevice():
+    # Check if given tap device is part of a network bridge
+    # if so use bridged interface instead of given tap device
+    tap = os.environ["TAPDEV"]
+    result = os.popen('bridge link | grep ' + tap)
+    bridge = re.search('master (.*) state', result.read())
+    
+    return bridge.group(1).strip() if bridge else tap
+
+def getHostLinkLocalAddr(interface):
+    result = os.popen('ip addr show dev ' + interface + ' scope link | grep inet6 ')
+    return re.search('inet6 (.*)/64', result.read()).group(1).strip()
+
+def generateHostPortNumber():
+    return random.randint(1024, 65535)
+
+def getRiotLinkLocalAddr(child):
```

Comment:
Maybe add `iface` to it, as you get the interface ID anyway.

Reply:
Clarify please

Reply:
Ok.. I did not understand, why you get the interface number, but not use it to get the ll-addr of a specific interface (using `ifconfig <if_id>`), but now looking at your code, I see that you don't need it, since you only have one interface and you use the interface ID to specify the interface a link-local address is from.